pkg/apiserver/middlewares/audit/audit.go

```diff
@@ -340,40 +340,31 @@ func (r responseBodyWriter) WriteString(s string) (n int, err error) {
 }
 
 func getPostObjectName(c *gin.Context) string {
-	// get request body
-	data, err := c.GetRawData()
-	if err != nil {
-		clog.Warn("[audit] get raw data err: %s", err.Error())
-		return ""
-	}
-	// put request body back
-	c.Request.Body = ioutil.NopCloser(bytes.NewBuffer(data))
+	obj := struct {
```

Comment:
How about using `unstructured.Unstructured` with couple of `xxx.GetName()` and `xxx.UnmarshalJSON()`, I just leave it for @caoyingjunz.

Reply:
> How about using `unstructured.Unstructured` with couple of `xxx.GetName()` and `xxx.UnmarshalJSON()`, I just leave it for @caoyingjunz.

@weilaaa done, thanks for advice

Reply:
@weilaaa please help review https://github.com/kubecube-io/KubeCube/pull/264